nemo/collections/multimodal/models/speechllm_models.py

```diff
@@ -433,10 +451,75 @@ def _build_dataset(self, data_cfg, is_train=True):
                 )
                 datasets.append(dataset)
 
-            dataset = BlendableDataset(
-                datasets=datasets, weights=concat_sampling_probabilities, size=num_train_samples_after_blend
+            if is_train:
+                dataset = BlendableDataset(
+                    datasets=datasets, weights=concat_sampling_probabilities, size=num_train_samples_after_blend
+                )
+                return dataset
+            else:
+                return datasets
+
+    def _build_tarred_dataset(self, data_cfg, augmentor):
+        return get_tarred_aqa_dataset_from_config(
+            config=data_cfg,
+            tokenizer=self.tokenizer,
+            augmentor=augmentor,
+            sep_id=self.sep_id,
+            answer_only_loss=self.cfg.get('answer_only_loss', True),
+            virtual_tokens=self.virtual_tokens,
+            global_rank=parallel_state.get_data_parallel_rank(),
+            world_size=parallel_state.get_data_parallel_world_size(),
+        )
+
+    def build_data_loader(self, dataset, data_cfg, consumed_samples=0):
+        """Buld dataloader given an input dataset."""
+        logging.info(f'Building dataloader with consumed samples: {consumed_samples}')
+        if isinstance(dataset, BlendableDataset):
+            collate_fn = dataset.datasets[0].collate_fn
+        elif hasattr(dataset, 'collate_fn'):
+            collate_fn = dataset.collate_fn
+        elif hasattr(dataset.datasets[0], 'collate_fn'):
+            # support datasets that are lists of entries
+            collate_fn = dataset.datasets[0].collate_fn
+        else:
+            # support datasets that are lists of lists
```

Comment:
could you explain why we need these for supporting tar?

Reply:
followed ASR tarred datasets, I think it's for bucketing